README.md

```diff
@@ -31,65 +20,105 @@ cm <- ConnectionManager({
+| ----------------- | ------------------- | -----                                                           |
+| cm                | N/A                 | A ConnectionManager object                                      |
+| flags             | `null`/`0`          | A bit array of binary options that control connection behaviour |
+| logger            | `server`            | A logger object with `.log()` and `.error()` methods            |
 
-# setLogs(logs)
+### Flags
 
-This function allows the user to change the logs object.
+Flags are boolean options passed as the 3rd argument to the constructor.  Multiple flags should be "added"/"stacked" with the binary "or" operator, `|`.  All flags are off/`false` by default.
 
-# setWifiList(wifiList)
+#### WIFIQUEUE\_HIDDEN
 
-This function allows the user to change the list of networks. When the device next attempts to connect it will start from the beginning of the list.
+This flag indicates whether any wifis listed in `wifiList` might be hidden.  If set, WifiQueue will try all networks in `wifiList` regardless of visibility.
 
-## connect()
+#### WIFIQUEUE\_OPEN
 
-This function triggers the device to start trying to connect. It will first attempt to connect to the network it was blinked up to and if that
-fails it will cycle through the list of networks provided until it succeeds.
+This flag indicates whether WifiQueue should try to connect to an visible
+networks that are open.
 
-Will return `false` if it is already trying to connect and `true` if it is already connected.
-If the current ssid is empty, the onFail handler will immediately be called.
+## Connection Management
+
+### connect(*wifis*)
+
+Triggers the device to start trying to connect.
+
+Will return `false` if it is already trying to connect and `true` if it is already connected.  If the current ssid is empty, the onFail handler will immediately be called.
+
+The list of wifis to connect to should be passed to `.connect()` on every call, but the array that's passed in will not be mutated.  This supports a wifi list that can change between connection attempts, but please *DO NOT* assume that calling `.connect()` without arguments will cause it to use any wifis that have been passed in previously.
 
 ```squirrel
-wq.connect();
+wifis <- [
+	{"ssid": "test1", "pw": "password"},
+	{"ssid": "test2", "pw": "password"},
+	{"ssid": "test3", "pw": "password"}
+]
+
+wq.connect(wifis);
 ```
 
-## disconnect()
+### disconnect()
 
-This function disconnects the device from WiFi. The `didDisconnect()` function will be triggered.
+Disconnects the device from WiFi. The `didDisconnect()` function will be triggered.
 
 ```squirrel
 wq.disconnect();
 ```
 
-## onConnect(*callback*)
+## Callback Setters
+
+### onConnect(*callback*)
 
-This function sets a callback to be trigger whenever the device connects.
+Sets a callback to be trigger whenever the device connects.
```

Comment:
Make sure to document that it will also be called if connect() is called when the device is already connected.